File: app/src/main/java/cn/zjl/datacollector/data/entity/WaveformDataEntity.java
package cn.zjl.datacollector.data.entity;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "data_sample")
public class WaveformDataEntity {

    @PrimaryKey(autoGenerate = true)
    public long id;

    public long startTime;

    public int deviceType;

    public int type;

    public int period;

    public byte[] dataRecv;

    public byte[] dataRecvPos;

    public byte[] dataRecvLen;

    public byte[] dataSend;

    public byte[] dataSoff;

    public float sendFs;

    public float simpleSendFs;

    public float simpleOffFs;

    public float recvFs;

    public int use;

    public String note;

    public long dataPointId;

    public long createdAt;
}

File: app/src/main/java/cn/zjl/datacollector/data/entity/WaveformDataEntity.kt
package cn.zjl.datacollector.data.entity

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "data_sample")
data class WaveformDataEntity(
    @PrimaryKey(autoGenerate = true) var id: Long = 0,
    var startTime: Long = 0,
    var deviceType: Int = 0,
    var type: Int = 0,
    var period: Int = 0,
    var dataRecv: ByteArray? = null,
    var dataRecvPos: ByteArray? = null,
    var dataRecvLen: ByteArray? = null,
    var dataSend: ByteArray? = null,
    var dataSoff: ByteArray? = null,
    var sendFs: Float = 0f,
    var simpleSendFs: Float = 0f,
    var simpleOffFs: Float = 0f,
    var recvFs: Float = 0f,
    var use: Int = 0,
    var note: String = "",
    var dataPointId: Long = 0,
    var createdAt: Long = 0
)
